app/reducers/builderUtils.ts

```diff
@@ -49,6 +51,58 @@ export function validateGrid(state: BuilderState) {
   };
 }
 
+export function getWarningStats(state: BuilderState) {
+  const shortWords = new Set<string>();
+
+  for (const [i, entry] of state.grid.entries.entries()) {
+    if (entry.cells.length <= 2) {
+      shortWords.add(entryWord(state.grid, i));
+    }
+  }
+
+  const unmatchedRefs = new Set<string>();
+  const missingEnums = new Set<string>();
+  const wrongEnums = new Set<string>();
+
+  const digitsRegex = /(\d+)/g;
+
+  const gridEntries = new Set<string>();
+  state.grid.entries.forEach((entry) => {
+    gridEntries.add(entryString(entry));
+  });
+
+  for (const [word, clues] of Object.entries(state.clues)) {
+    for (const clue of clues) {
+      const refs = parseClueReferences(clue);
+      if (!refs.every((ref) => gridEntries.has(entryString(ref)))) {
+        unmatchedRefs.add(word);
+      }
+
+      const clueEnum = parseClueEnumeration(clue);
+      if (clueEnum == null) {
+        missingEnums.add(word);
+      } else {
+        const length = clueEnum.match(digitsRegex)?.reduce((sum, str) => {
+          const val = +str;
+          return isNaN(val) ? sum : sum + val;
+        }, 0);
+        if (length == null || length !== word.length) {
+          wrongEnums.add(word);
+        }
+      }
+    }
+  }
+  const enumsExpected = missingEnums.size < Object.keys(state.clues).length;
```

Comment:
I wonder if we should fuzz this a bit more - like if anything less than half of the clues have enums then the constructor was doing something else w/ the parens and we shouldn't be expecting enums at all? If I'm reading this correctly then a single clue which ends with a parenthetical that happens to contain a digit would trigger a missing enums and a wrong enums. Certainly in meta puzzles it's kind of common to end some number of clues w/ an enum looking thing - see here for a puzzle that ideally wouldn't trigger any enum warnings: https://crosshare.org/crosswords/iUvSEeMaNMabtBeEhSAj/heart-of-gold-meta

Reply:
Ah, this was a goof! Meant to divide by two 🤦